packages/mermaid/src/docs/community/development.md

```diff
@@ -1,77 +1,151 @@
 
-The documentation is located in the `src/docs` directory and organized according to relevant subfolder.
+**Ex: A new feature described in issue 2945 that adds a new arrow type called 'florbs' to state diagrams**
 
-The `docs` folder will be automatically generated when committing to `src/docs` and should not be edited manually.
+`feature/2945_state-diagram-new-arrow-florbs`
+
+**Ex: A bug described in issue 1123 that causes random ugly red text in multiple diagram types**
+`bug/1123_fix_random_ugly_red_text`
+
+### 2. Write Tests
+
+Tests ensure that each function, module, or part of code does what it says it will do. This is critically
+important when other changes are made to ensure that existing code is not broken (no regression).
 
-We encourage contributions to the documentation at [mermaid-js/mermaid/src/docs](https://github.com/mermaid-js/mermaid/tree/develop/packages/mermaid/src/docs). We publish documentation using GitHub Pages with [Docsify](https://www.youtube.com/watch?v=TV88lp7egMw&t=3s)
+Just as important, the tests act as _specifications:_ they specify what the code does (or should do).
+Whenever someone is new to a section of code, they should be able to read the tests to get a thorough understanding of what it does and why.
 
-### Add Unit Tests for Parsing
+If you are fixing a bug, you should add tests to ensure that your code has actually fixed the bug, to specify/describe what the code is doing, and to ensure the bug doesn't happen again.
+(If there had been a test for the situation, the bug never would have happened in the first place.)
+You may need to change existing tests if they were inaccurate.
 
-This is important so that, if someone that does not know about this great feature suggests a change to the grammar, they get notified early on when that change breaks the parser. Another important aspect is that, without proper parsing, tests refactoring is pretty much impossible.
+If you are adding a feature, you will definitely need to add tests. Depending on the size of your feature, you may need to add integration tests.
 
-### Add E2E Tests
+#### Unit Tests for Parsing
 
-This tests the rendering and visual appearance of the diagrams. This ensures that the rendering of that feature in the e2e will be reviewed in the release process going forward. Less chance that it breaks!
+If you are adding or changing the text that describes a diagram (the _grammar_), you will need to add (or change) tests for the _parser._
+
+#### Integration/End-to-End (e2e) tests
+
+These test the rendering and visual appearance of the diagrams.
+This ensures that the rendering of that feature in the e2e will be reviewed in the release process going forward. Less chance that it breaks!
 
 To start working with the e2e tests:
 
-1. Run `pnpm run dev` to start the dev server
-2. Start **Cypress** by running `pnpm exec cypress open` in the **mermaid** folder.
+1. Run `pnpm run dev` to start the dev server (or use the `pnpm dev` script)
```

Comment:
Isn't `pnpm run dev` and `pnpm dev` the same?

Reply:
yup. :-)